### PR TITLE
msbuild: fix overbuilding of externals and lessen rebuild likelyhood

### PR DESCRIPTION
**Pull request by shuffle2**
This partially reverts #9092 (which created `Externals\ExternalsReferenceAll.props`)
Instead, Externals now "publish" `exports.props` files that referencing projects can import in order to 1) reference the project (for dependency graph) and 2) include any `ClCompile` settings (+etc) exported by the external (used for include directories and definitions).

This also splits some of `Base.props` out into a seperate file, such that `Base.props` is now only for truly global settings, whereas the new file is for settings shared between dolphin-authored code (but not Externals).

While working on PRs recently, I've noticed that msbuild would sometimes overbuild after making small edits. This PR _should_ fix that (please let me know if you ever see such a thing). It also makes rebuilds of Externals less likely, as `Base.props` does not need to be edited as frequently. The new behavior is also more correct in the sense that include paths and definitions for all External libraries are not broadcast into everything (fun fact: this was extremely vestigial dolphin behavior, based on how SCons was setup, when dolphin used that).

For full rebuilds, there is a slight perf regression for machines with a lot of cpu cores (like, > 32). However, with some tips from https://devblogs.microsoft.com/cppblog/cpp-build-throughput-investigation-and-tune-up speed can be regained if needed. I also realized most people building dolphin don't have such machines, and would rather use less resources by default (oops..)

### CODE REVIEW

**Comment by Rumi-Larry** [comment]
Is there a reason some GUIDs are lowercase and others lowercase?

**Review by phire — APPROVED** [review]

File: Source/Core/DolphinQt/DolphinQt.vcxproj [hunk 1]
```diff
@@ -403,7 +404,7 @@
       <Project>{0e033be3-2e08-428e-9ae9-bc673efa12b5}</Project>
     </ProjectReference>
     <!--
-      This project doesn't uses its own PCH during compile (because RTTI setting differs),
+      This project uses its own PCH during compile (because RTTI setting differs),
       but we must also link the "main" pch for the dependants (DolphinLib)
     -->
     <ProjectReference Include="$(SourceDir)PCH\pch.vcxproj">
```
Comment (Pokechu22) [inline Source/Core/DolphinQt/DolphinQt.vcxproj:410]: I assume this was a typo in #10772?

File: Externals/imgui/imgui.vcxproj [hunk 2]
```diff
@@ -15,6 +15,12 @@
     <Import Project="$(VSPropsDir)ClDisableAllWarnings.props" />
   </ImportGroup>
   <PropertyGroup Label="UserMacros" />
+  <ItemDefinitionGroup>
+    <ClCompile>
+      <!--IM_ASSERT is defined as Dolphin's ASSERT() :( (DolphinLib circular dependency)-->
+      <AdditionalIncludeDirectories>$(CoreDir);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+    </ClCompile>
+  </ItemDefinitionGroup>
```
Comment (Pokechu22) [inline Externals/imgui/imgui.vcxproj:23]: See #10375. Is this actually a circular dependency? I don't think common depends directly on imgui, so it should still be acyclic (though it is still a library depending on Dolphin).

File: Externals/liblzma/liblzma.vcxproj [hunk 3]
```diff
@@ -17,7 +17,7 @@
   <PropertyGroup Label="UserMacros" />
   <ItemDefinitionGroup>
     <ClCompile>
-      <AdditionalIncludeDirectories>.;check;common;delta;lz;lzma;rangecoder;simple;tuklib;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>.;api;check;common;delta;lz;lzma;rangecoder;simple;tuklib;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
       <PreprocessorDefinitions>LZMA_API_STATIC;HAVE_CONFIG_H;%(PreprocessorDefinitions)</PreprocessorDefinitions>
     </ClCompile>
   </ItemDefinitionGroup>
```
Comment (Pokechu22) [inline Externals/liblzma/liblzma.vcxproj:23]: What's the deal with these changes? Were these incorrect beforehand?

File: Source/Core/DolphinQt/DolphinQt.vcxproj (hunk 1 above)
Comment (shuffle2) [reply Source/Core/DolphinQt/DolphinQt.vcxproj:410]: initially when moving to Qt6 i disabled pch on DolphinQt and added the original comment. Then I enabled pch for DolphinQt and modified the comment, and made a typo.

File: Externals/imgui/imgui.vcxproj (hunk 2 above)
Comment (shuffle2) [reply Externals/imgui/imgui.vcxproj:23]: i guess it depends on your viewpoint. technically i dont think it's circular, but it is not great to have externals including dolphin code

File: Externals/liblzma/liblzma.vcxproj (hunk 3 above)
Comment (shuffle2) [reply Externals/liblzma/liblzma.vcxproj:23]: previously this path was added by Base.props, which was the wrong place to put it
`<AdditionalIncludeDirectories>$(ExternalsDir)liblzma\api;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>`

Comment (Pokechu22) [reply Externals/liblzma/liblzma.vcxproj:23]: Was the original goal to have some headers be public and some be private? I know visual studio seems to include zstd's `common` folder (containing e.g. `common/bitstream.h`) in the list of headers for intellisense (and it prioritizes that capitalization over dolphin's Common, though both show all files). If that was the goal originally, it didn't work, but it still seems like something that would be good to do.

Comment (shuffle2) [reply Externals/liblzma/liblzma.vcxproj:23]: Base.props was only supposed to have the "public" headers, so this specific case (liblzma\api) was just incorrect.

The priority for build is based on order the arguments are passed on the commandline, which is a direct result of how AdditionalIncludeDirectories is formed in the xml.

**Comment by AdmiralCurtiss** [comment]
I'll just believe you that this is correct.

**Pull request merged**